dragonfly/implementations/PsyllidProvider.py

```diff
@@ -0,0 +1,404 @@
+
+        
+    def get_all_central_frequencies(self):
+        return self.freq_dict
+    
+
+    def set_central_frequency(self, channel, cf):
+        logger.info('Trying to set cf of channel {} to {}'.format(channel, cf))
+        try:
+            if self.mode_dict[channel] == 'streaming':
+                request = '.active-config.ch'+str(self.channel_dict[channel])+'.strw.center-freq'
+                result = self.provider.set(self.queue_dict[channel]+request, cf)
+                logger.info('Set central frequency of streaming writer for channel {} to {} Hz'.format(channel, cf))
+                self.freq_dict[channel]=cf
+                return True
+
+            elif self.mode_dict[channel] == 'triggered':
+                request = 'active-config.ch'+str(self.channel_dict[channel])+'.trw.center-freq'
+                result = self.provider.set(self.queue_dict[channel]+'.'+request, cf)
+                logger.info('Set central frequency of egg writer for channel {} to {} Hz'.format(channel, cf))
+                self.freq_dict[channel]=cf
+                return True
+        except:
+            logger.error('Could not set central frequency')
+            self.freq_dict[channel]=None
+            return False
+
+
+    def start_run(self, channel, duration, filename):
+        if duration <= 10:
+            return False
+        payload = {'duration':duration, 'filename':filename}
+        result = self.provider.cmd(self.queue_dict[channel], 'start-run', payload=payload)
+        return True
+
+
+    def stop_run(self, channel):
+        result = self.provider.cmd(self.queue_dict[channel], 'stop-run')
+
+
+    def get_number_of_channels(self):
+        active_channels = [i for i in self.freq_dict.keys() if self.freq_dict[i]!=None]
+        logger.info('Active channels are {}'.format(active_channels))
+        return len(active_channels)
+
+
+    def get_active_channels(self):
+        active_channels = [i for i in self.freq_dict.keys() if self.freq_dict[i]!=None]
+        return active_channels
+
+
+    def get_number_of_streams(self, channel):
+        stream_count = 0
+        cf = self.freq_dict[channel]
+        for i in range(3):
+            try:
+                request = '.node-config.ch'+str(i)+'.strw.center-freq'
+                result = self.provider.set(self.queue_dict[channel]+request, cf)
+                stream_count += 1
+                self.mode_dict[channel]='streaming'
+            except:
+                try:
+                    request = '.node-config.ch'+str(i)+'.trw.center-freq'
+                    result = self.provider.set(self.queue_dict[channel]+request, cf)
+                    stream_count += 1
+                    self.mode_dict[channel]='triggered'
+                except:
+                    pass  
+        logger.info('Number of streams for channel {}: {}'.format(channel, stream_count))
+        return stream_count
+
+
+    def set_fmt_snr_threshold(self, channel='a', snr=6):
```

Comment:
As with get_* above, you can use `@<property>.setter` to make attributes which have function implementations. This is more pythonic and as above, will allow access via `dragonfly get` and `dragonfly set`.

For example, if the "tau" property is defined as 2x the "pi" property, you could do:
```
@property
def tau(self):
    return self.pi*2.
@tau.setter
def tau(self, value)
    self.pi = value/2.
```